### ISSUE TITLE
Adding a possibility to create mirrors.

### ISSUE THREAD

**Issue by Andrey2470T** [minetest/minetest#7732]
##### Issue type

 - Feature request
 
Hi. Could you, please, add a possibility to create mirrors in minetest? I mean real mirror. Thanks!

**Comment by rubenwardy** [comment]
Duplicate of #7328 #6996

**Comment by rubenwardy** [comment]
Surprisingly, I can't find an issue to make this a duplicate. Could have sworn there was an issue for this

**Comment by paramat** [comment]
:-1: seems not worth the effort, sorry.

**Comment by rubenwardy** [comment]
I think that this feature should be added eventually, but we need someone with graphics expertise to fix more important issues and add greater stability first. *cough* z-sorting *cough*

**Comment by Andrey2470T** [comment]
is this feature planned to be added for 5.0.0 or later?

**Comment by rubenwardy** [comment]
No :P

**Comment by Andrey2470T** [comment]
:(((

**Comment by HybridDog** [comment]
> […] little usefulness, and unsuitable […]

I think there are many use cases:

Reflecting mirrors could be used for [the laser beam mod](https://github.com/hybriddog/laser).

Water could look a lot more like water if there were reflections, an animated normalmap could be used to control the reflection angles.

The steel block currently doesn't look metallic because of missing reflections:
![out](https://user-images.githubusercontent.com/3192173/45899577-84e7b580-bddd-11e8-9a3d-c762eac4aac7.png)
For comparison, a screenshot of metal textures with reflection in sauerbraten:
![out](https://user-images.githubusercontent.com/3192173/45899833-456d9900-bdde-11e8-837b-373461ed7d21.jpg)

**Comment by paramat** [comment]
By 'usefulness' i mean actual gameplay benefits other than visual decoration. If you try of course you can think of uses, but they are low priority and specialised.
MT is not a photorealistic voxel game engine, it is for MC type abstraction, where photorealism usually looks weird and out of place in a world made of cubes. So reflecting water and metal is low priority and non-essential.
Anyone who wants photorealism would not be coming to MT for it because MT is one of the least photorealistic engines around. It would be ridiculous to try to be photorealistic and to be every type of game engine, specialisation is necessary.

**Comment by rubenwardy** [comment]
Reflection is barely photorealistic level

**Comment by benrob0329** [comment]
Cubemap based reflections have been a thing long before photorealistic gaming was viable. Heck, we even have a cubemap (the sky) that we could base reflections off of.

**Comment by HybridDog** [comment]
> [It] usually looks weird and out of place in a world made of cubes.

Of course it should configurable with a setting, if you personally do not like it, you can simply disable it in the settings tab.
Lots of voxel games have water reflection:
![baw](https://www.buildaworld.net/sites/all/themes/baw/images/gallery/showme/37.jpg)
![terasology](https://www.freeware.de/images/screenshot_slideshow/terasology-screenshot-54e80daff007d.jpeg)

**Comment by Ezhh** [comment]
Improvements to water appearance are one of my greatest wishes for MT, and have been for a very long time. Reflections would simply be wonderful to have.

**Comment by paramat** [comment]
Yeah thinking on this reflection and shinyness is valid in an abstract world. Not high priority though.
> Of course it should configurable with a setting, if you personally do not like it, you can simply disable it in the settings tab.

This is the invalid 'we should have everything because it can be an option' argument.

**Comment by rubenwardy** [comment]
That is just wrong, Paramat. It doesn't half FPS because you don't send the data to the GPU twice

**Comment by paramat** [comment]
Still a very significant load though per mirror surface?
Each new camera has to do the intensive rendering of the view from the view frustrum intersecting meshes. It probably isn't twice the load but i expect not far off. I realise some stuff isn't doubled in intensity.

**Comment by rubenwardy** [comment]
Very unlikely to be significant on newer GPUs

**Comment by benrob0329** [comment]
We don't need full screen space reflections to start with, cubemap reflections based on the sky would be more than enough for now IMO.
